monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py

```diff
@@ -0,0 +1,52 @@
+from jsonschema import validate
+
+from common.agent_configuration import AgentConfiguration
+from monkey_island.cc.repositories import (
+    IAgentConfigurationRepository,
+    PluginConfigurationValidationError,
+    RetrievalError,
+)
+from monkey_island.cc.repositories.utils import AgentConfigurationSchemaCompiler
+
+
+class AgentConfigurationValidationDecorator(IAgentConfigurationRepository):
+    """
+    A IAgentConfigurationRepository decorator that validates the agent configuration
+    """
+
+    def __init__(
+        self,
+        agent_configuration_repository: IAgentConfigurationRepository,
+        agent_configuration_schema_compiler: AgentConfigurationSchemaCompiler,
+    ):
+        self._agent_configuration_repository = agent_configuration_repository
+        self._agent_configuration_schema_compiler = agent_configuration_schema_compiler
+
+    def get_configuration(self) -> AgentConfiguration:
+        try:
+            agent_configuration_schema = self._agent_configuration_schema_compiler.get_schema()
+            agent_configuration = self._agent_configuration_repository.get_configuration()
```

Comment:
This is not a decorator it's a wrapper, because it applies only to this method. Might as well just call it "get_valid_configuration" and do this in the repository

Reply:
How it is just only applying to this method when we validate the configuration before we update?

Reply:
"Wrapper" is another name for "decorator". Please see [this](https://refactoring.guru/design-patterns/decorator) for a discussion of the decorator pattern.